### ISSUE TITLE
Should there be an error when not awaiting in an async function?

### ISSUE THREAD

**Issue by tzachshabtay** [palantir/tslint#3225]
### Not sure if this is a bug report or feature request

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.3.4
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
// code snippet
export module XXX {
    export async function YYY(): Promise<void> {
           //any synchronous code only goes here (or just an empty function, reproduces too)
    }
}
```

with `tslint.json` configuration:

```json
{
  "lintOptions": {
    "typeCheck": true
  },
  "rules": {
    "interface-name": false,
    "interface-over-type-literal": false,
    "max-classes-per-file": false,
    "member-ordering": [
      true,
      "public-before-private",
      "static-before-instance",
      "variables-before-functions"
    ],
    "no-angle-bracket-type-assertion": true,
    "no-default-export": true,
    "no-empty-interface": true,
    "no-string-throw": true,
    "no-switch-case-fall-through": true,
    "prefer-const": true,
    "switch-default": true,
    "variable-name": [
      true,
      "ban-keywords",
      "check-format",
      "allow-leading-underscore",
      "allow-pascal-case"
    ],
    "typedef": [
      true,
      "call-signature",
      "arrow-call-signature",
      "parameter",
      "property-declaration",
      "variable-declaration",
      "member-variable-declaration",
      "object-destructuring",
      "array-destructuring"
    ],
    "indent": [
      true,
      "spaces"
    ],
    "cyclomatic-complexity": true,
    "no-trailing-whitespace": true,
    "semicolon": true
  }
}
```

#### Actual behavior

This compiles fine.

#### Expected behavior

Expecting to see some error like: `Async function must have an await inside it.`

**Comment by ajafff** [comment]
Currently there is no rule to enforce this. So this is a feature request / rule suggestion.

Does it really make sense to enforce this? You can also use `async` to convert sync exceptions into rejected promises:

```ts
async function onlyReturnsPromise() {
    return throwsOrReturnsPromise();
}
```

**Comment by JoshuaKGoldberg** [comment]
Seems like this rule isn't in high demand. If you have a pressing need for it, please do post here and we can re-open!
